spec/models/weight_limits_spec.rb

```diff
@@ -91,6 +101,171 @@ def build_inventory_unit(variant, order)
       Spree::ActiveShipping::Config.set(:default_weight => 1)
     end
 
+    describe ".valid_weight_for_package?" do
```

Comment:
I agree with you on all of this. I think if we can extract all of the package weight stuff into a class without modifying the logic and adding the tests you have for that. We'll be in a much better place and avoid testing these private methods.

Let me know if you want to chat more about what that might look like. Thanks again for going through and figuring out all this convoluted logic :heart: